frontend/src/component/admin/apiToken/ApiTokenPage/ApiTokenPage.tsx

```diff
@@ -1,19 +1,93 @@
-import { useContext } from 'react';
+import { useContext, useMemo } from 'react';
 import AccessContext from 'contexts/AccessContext';
 import { ConditionallyRender } from 'component/common/ConditionallyRender/ConditionallyRender';
-import { READ_API_TOKEN } from 'component/providers/AccessProvider/permissions';
+import {
+    CREATE_API_TOKEN,
+    DELETE_API_TOKEN,
+    READ_API_TOKEN,
+} from 'component/providers/AccessProvider/permissions';
 import { AdminAlert } from 'component/common/AdminAlert/AdminAlert';
-import { ApiTokenTable } from 'component/admin/apiToken/ApiTokenTable/ApiTokenTable';
+import { ApiTokenTable } from 'component/common/ApiTokenTable/ApiTokenTable';
+import { PageContent } from 'component/common/PageContent/PageContent';
+import { PageHeader } from 'component/common/PageHeader/PageHeader';
+import { CreateApiTokenButton } from 'component/common/ApiTokenTable/CreateApiTokenButton/CreateApiTokenButton';
+import { Search } from 'component/common/Search/Search';
+import { useApiTokenTable } from 'component/common/ApiTokenTable/useApiTokenTable';
 import { useApiTokens } from 'hooks/api/getters/useApiTokens/useApiTokens';
+import { ActionCell } from 'component/common/Table/cells/ActionCell/ActionCell';
+import { CopyApiTokenButton } from 'component/common/ApiTokenTable/CopyApiTokenButton/CopyApiTokenButton';
+import { RemoveApiTokenButton } from 'component/common/ApiTokenTable/RemoveApiTokenButton/RemoveApiTokenButton';
+import useApiTokensApi from 'hooks/api/actions/useApiTokensApi/useApiTokensApi';
 
 export const ApiTokenPage = () => {
     const { hasAccess } = useContext(AccessContext);
-    const { tokens, loading } = useApiTokens();
+    const { tokens, loading, refetch } = useApiTokens();
+    const { deleteToken } = useApiTokensApi();
+
+    const {
+        getTableProps,
+        getTableBodyProps,
+        headerGroups,
+        rows,
+        prepareRow,
+        state: { globalFilter },
+        setGlobalFilter,
+        setHiddenColumns,
+        columns,
+    } = useApiTokenTable(tokens, props => (
+        <ActionCell>
+            <CopyApiTokenButton
+                token={props.row.original}
+                permission={READ_API_TOKEN}
+            />
+            <RemoveApiTokenButton
+                token={props.row.original}
+                permission={DELETE_API_TOKEN}
+                remove={async () => {
```

Comment:
onRemove to follow onSth convention?

Reply:
It's called onRemove inside the RemoveApiToken button, the convention is meant for functions that are passed directly as event handlers.

Reply:
Fair enough, I'll fix